src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -469,6 +469,21 @@ public void replace(CtElement element) {
 		ReplacementVisitor.replace(this, elements);
 	}
 
+	@Override
+	public <E extends CtElement> E setAllMetadata(Map<String, Object> metadata) {
+		if (metadata == null || metadata.isEmpty()) {
+			this.metadata = new HashMap<>();
```

Comment:
empty hashmap has some size in memory and if we would have such empty hash maps more often (for each CtElement) then Spoon model would be n-times bigger - for nothing. 

I would prefer to use null to represent "no metadata attribute". WDYT?